src/lavinmq/amqp/client.cr

```diff
@@ -413,16 +413,20 @@ module LavinMQ
       private def close_socket
         @running = false
         @socket.close
-        @log.debug { "Socket closed" }
       rescue ex
         @log.debug { "#{ex.inspect} when closing socket" }
       end
 
-      def close(reason = nil)
+      def close(reason = nil, timeout : Time::Span = 60.seconds)
         reason ||= "Connection closed"
-        @log.info { "Closing, #{reason}" }
-        send AMQP::Frame::Connection::Close.new(320_u16, "CONNECTION_FORCED - #{reason}", 0_u16, 0_u16)
-        @running = false
+        @log.info { "Closing: #{reason}" }
+
+        socket = @socket
+        if socket.responds_to?(:"write_timeout=")
+          socket.write_timeout = timeout
+        end
+
+        socket.close
```

Comment:
we're not sending a Close::Connection frame anymore?

Reply:
We are locally on my computer. I guess i forgot to push that commit 🙃